### PR TITLE
Add support for Intel Gaudi Profiler.

### PR DESCRIPTION
**Pull request by ankitgola005**
<details>
  <summary><b>Add support for Intel Gaudi Profiler. </b></summary>

- [ ] Was this discussed/agreed via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>

## What does this PR do?
Remove env var from HPUProfiler. Add test.

Fixes # (issue).

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in GitHub issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/test_pytorch/test_profiler.py [hunk 1]
```diff
@@ -280,3 +280,9 @@ def test_hpu_trace_event_kernel(tmpdir):
             raise Exception("Could not find event kernel in trace")
         for event_duration in event_duration_arr:
             assert event_duration >= 0
+
+
+def test_hpu_profiler_env(monkeypatch):
```
Comment (jerome-habana) [inline tests/test_pytorch/test_profiler.py:285]: is there any profiler test that we need to enable back ?

File: src/lightning_habana/pytorch/profiler/profiler.py [hunk 2]
```diff
@@ -89,7 +89,10 @@ def __init__(
         record_module_names: bool = True,
         **profiler_kwargs: Any,
     ) -> None:
-        os.environ["HABANA_PROFILE"] = "1"
+        assert os.environ.get("HABANA_PROFILE", None) in (
+            None,
+            "profile_api_light",
+        ), "`HABANA_PROFILE` should not be set when using `HPUProfiler`"
```
Comment (jerome-habana) [inline src/lightning_habana/pytorch/profiler/profiler.py:95]: add this to documentation

File: tests/test_pytorch/test_profiler.py (hunk 1 above)
Comment (ankitgola005) [reply tests/test_pytorch/test_profiler.py:284]: enabled

File: docs/source/advanced.rst [hunk 3]
```diff
@@ -124,6 +124,20 @@ Limitations
...
+Using Intel Gaudi Profiler
+------------------------------
+
+The Intel Gaudi Profiling subsystem, and the Profiling Configuration tools are methods to configure Intel Gaudi Profiler.
+
+Please refer to `Getting Started with Intel Gaudi Profiler <https://docs.habana.ai/en/latest/Profiling/Intel_Gaudi_Profiling/Getting_Started_with_Profiler.html>`_ for more information.
+
+.. note::
+
+    ``HPUProfiler`` and Intel Gaudi Profiler should not be used together. Therefore, `HABANA_PROFILE` should not be set in environment when using ``HPUProfiler``.
```
Comment (jerome-habana) [inline docs/source/advanced.rst:138]: will hpuprofiler work now ?

Comment (ankitgola005) [reply docs/source/advanced.rst:149]: yes.  This value is set to "profile_api_light" internally to enable profiler.

File: docs/source/advanced.rst [hunk 4]
```diff
@@ -126,6 +133,30 @@ Limitations
 
 ----
 
+Using Intel Gaudi Profiler
+------------------------------
```
Comment (Borda) [inline docs/source/advanced.rst:137]:
```suggestion
--------------------------
```

**Review by Borda — APPROVED** [review]

**Review by jerome-habana — APPROVED** [review]

**Pull request merged**